include/bm/bm_sim/packet.h

```diff
@@ -105,7 +105,7 @@ class Packet final {
   using buffer_state_t = PacketBuffer::state_t;
 
   //! Number of general purpose registers per packet
-  static constexpr size_t nb_registers = 2u;
+  static constexpr size_t nb_registers = 3u;
```

Comment:
could we make it 4?
a very good reason: I don't like 3
a not so good reason: reserve one register for future use

Reply:
no problem